pandas/core/frame.py

```diff
@@ -3960,52 +3959,40 @@ def f(i):
 
     def _combine_series(self, other, func, fill_value=None, axis=None,
                         level=None, try_cast=True):
+        if fill_value is not None:
+            raise NotImplementedError("fill_value %r not supported." %
+                                      fill_value)
+
         if axis is not None:
             axis = self._get_axis_name(axis)
             if axis == 'index':
-                return self._combine_match_index(other, func, level=level,
-                                                 fill_value=fill_value,
-                                                 try_cast=try_cast)
+                return self._combine_match_index(other, func, level=level)
             else:
                 return self._combine_match_columns(other, func, level=level,
-                                                   fill_value=fill_value,
                                                    try_cast=try_cast)
-        return self._combine_series_infer(other, func, level=level,
-                                          fill_value=fill_value,
-                                          try_cast=try_cast)
+        else:
+            if len(other) == 0:
+                return self * np.nan
 
-    def _combine_series_infer(self, other, func, level=None,
-                              fill_value=None, try_cast=True):
-        if len(other) == 0:
-            return self * np.nan
+            if len(self) == 0:
+                # Ambiguous case, use _series so works with DataFrame
```

Comment:
what is this comment?

Reply:
Not totally sure.  Possibly a holdover from higher-dim implementation?

Reply:
Blame shows that this comment was introduced in c35a39de6a29a519f9e0 in 2011.  Looks like it replaced a comment `# Ambiguous case, use _series so works with DataMatrix`.  @wesm is this still relevant?  Suggestions for fleshing this out?